File: FurnitureStore/FurnitureStore.Models/GiftCard/GiftCard.cs

namespace FurnitureStore.Models.GiftCard
{
    public class GiftCard
    {
        public long Id { get; set; }
        public string Name { get; set; }
        public string CardNumber { get; set; }
        public int Amount { get; set; }
        public DateTime ExpiryDate { get; set; }
        public bool IsActivated { get; set; }
        public long? ImageId { get; set; }
        public string? ImagePath { get; set; }
    }
}

File: FurnitureStore/UI/furniturestore_mobile/lib/models/gift_card/gift_card.dart
import 'package:json_annotation/json_annotation.dart';

part 'gift_card.g.dart';

@JsonSerializable()
class GiftCardModel {
  int? id;
  String? name;
  String? cardNumber;
  int? amount;
  DateTime? expiryDate;
  bool? isActivated;
  String? imagePath;
  int? imageId;

  GiftCardModel({
    this.id,
    this.name,
    this.cardNumber,
    this.amount,
    this.expiryDate,
    this.isActivated,
    this.imagePath,
    this.imageId,
  });

  factory GiftCardModel.fromJson(Map<String, dynamic> json) =>
      _$GiftCardModelFromJson(json);

  Map<String, dynamic> toJson() => _$GiftCardModelToJson(this);
}
